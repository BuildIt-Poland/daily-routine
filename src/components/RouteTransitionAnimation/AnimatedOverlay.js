import React from 'react';
import PropTypes from 'prop-types';
import styled, { withTheme } from 'styled-components';
import { animated, useSpring } from 'react-spring';

import { theme } from '../../types';
import { colorWhite, zIndexModalOverlay } from '../../styles/designTokens';

const Wrapper = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: ${zIndexModalOverlay};
  overflow: hidden;

  /* Show app bar on desktop and tablet when window size is higher than whole character artwork */
  @media (min-height: 51.25em) and (min-width: 31.25em) {
    top: 6rem;
  }
`;

const FullScreenCurtain = styled(animated.div)`
  width: 100%;
  height: 160vh;
  background-color: ${colorWhite};
  left: 0;
  position: absolute;
  z-index: ${zIndexModalOverlay};
  opacity: 1;
  transform: scaleY(0) translateY(0);
  transform-origin: 0 0;
`;

const FullScreenBar = styled(FullScreenCurtain)`
  z-index: ${zIndexModalOverlay + 2};
  transform-origin: 100% 100%;
`;

const ThickBar = styled(animated.div)`
  width: 100%;
  height: 0;
  left: 0;
  position: absolute;
  z-index: ${zIndexModalOverlay + 3};
  transform: translateY(0);
`;

function AnimatedOverlay({ onFinished, theme }) {
  const { primaryColor } = theme;

  const fullScreenCurtainAnimation = useSpring({
    from: {
      transform: 'scaleY(0)'
    },
    to: async next => {
      await next({ transform: 'scaleY(1)' });
      await next({ transform: 'scaleY(0)' });
    },
    config: { mass: 2, tension: 160, friction: 22, precision: 0.1, clamp: true },
    onRest: onFinished
  });

  const fullScreenBarAnimation = useSpring({
    from: {
      transform: 'scaleY(0) translateY(50%)',
      opacity: 1,
      backgroundColor: primaryColor
    },
    to: {
      transform: 'scaleY(1) translateY(0%)',
      opacity: 0,
      backgroundColor: primaryColor
    },
    config: { tension: 140, friction: 24, clamp: true }
  });

  const thickBarAnimation = useSpring({
    from: {
      height: '0%',
      transform: 'translateY(-20vh)',
      backgroundColor: primaryColor
    },
    to: {
      height: '30%',
      transform: 'translateY(160vh)',
      backgroundColor: primaryColor
    },
    config: { tension: 300, friction: 140 }
  });

  return (
    <Wrapper>
      <FullScreenCurtain style={fullScreenCurtainAnimation} />
      <ThickBar style={thickBarAnimation} />
      <FullScreenBar style={fullScreenBarAnimation} />
    </Wrapper>
  );
}

AnimatedOverlay.propTypes = {
  onFinished: PropTypes.func.isRequired,
  theme: theme.isRequired
};

export default withTheme(AnimatedOverlay);
