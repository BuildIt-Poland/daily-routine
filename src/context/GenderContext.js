import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

import { MALE } from '../constants/genders';

const GenderContext = createContext();
const { Provider } = GenderContext;

function GenderProvider({ children }) {
  const handleGenderChange = gender => setGender(gender);

  const [gender, setGender] = useState(MALE);

  return <Provider value={{ gender, handleGenderChange }}>{children}</Provider>;
}

GenderProvider.propTypes = {
  children: PropTypes.node
};

export { GenderContext, GenderProvider };
