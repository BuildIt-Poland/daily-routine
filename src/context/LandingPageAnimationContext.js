import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

const LandingPageAnimationContext = createContext();
const { Provider } = LandingPageAnimationContext;

function LandingPageAnimationProvider({ children, history }) {
  const REDIRECT_DELAY = 1000;

  function redirectWithDelay(url) {
    setTimeout(() => {
      history.push(url);
    }, REDIRECT_DELAY);
  }

  function animateAndRedirect(redirectUrl) {
    toggleAnimation(state => ({
      ...state,
      isAnimated: true
    }));
    redirectWithDelay(redirectUrl);
  }

  function stopAnimation() {
    toggleAnimation(state => ({
      ...state,
      isAnimated: false
    }));
  }

  const initialState = {
    isAnimated: false,
    animateAndRedirect,
    stopAnimation
  };

  const [animation, toggleAnimation] = useState(initialState);

  return <Provider value={animation}>{children}</Provider>;
}

LandingPageAnimationProvider.propTypes = {
  children: PropTypes.node,
  history: PropTypes.object
};

export { LandingPageAnimationContext, LandingPageAnimationProvider };
