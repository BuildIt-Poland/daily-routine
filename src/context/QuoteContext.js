import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

const QuoteContext = createContext();
const { Provider } = QuoteContext;

function QuoteProvider({ children }) {
  const [quote, setQuote] = useState(null);

  const handleQuoteChange = quote => setQuote(quote);

  return <Provider value={{ quote, handleQuoteChange }}>{children}</Provider>;
}

QuoteProvider.propTypes = {
  children: PropTypes.node
};

export { QuoteContext, QuoteProvider };
