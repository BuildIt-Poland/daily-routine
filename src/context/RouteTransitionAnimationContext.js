import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

import { history } from '../types';

const RouteTransitionAnimationContext = createContext();
const { Provider } = RouteTransitionAnimationContext;

function RouteTransitionAnimationProvider({ children, history }) {
  const REDIRECT_DELAY = 250;

  function redirectWithDelay(url) {
    setTimeout(() => {
      history.push(url);
    }, REDIRECT_DELAY);
  }

  function animateAndRedirect(redirectUrl) {
    setIsAnimating(true);
    redirectWithDelay(redirectUrl);
  }

  function stopAnimation() {
    setIsAnimating(false);
  }

  const [isAnimating, setIsAnimating] = useState(false);

  return <Provider value={{ isAnimating, animateAndRedirect, stopAnimation }}>{children}</Provider>;
}

RouteTransitionAnimationProvider.propTypes = {
  children: PropTypes.node,
  history: history.isRequired
};

export { RouteTransitionAnimationContext, RouteTransitionAnimationProvider };
