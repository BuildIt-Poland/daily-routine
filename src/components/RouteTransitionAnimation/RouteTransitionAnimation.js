import React, { useContext } from 'react';

import { RouteTransitionAnimationContext } from '../../context/RouteTransitionAnimationContext';
import useLockBodyScroll from '../../hooks/useLockBodyScroll';
import AnimatedOverlay from './AnimatedOverlay';

function RouteTransitionAnimation({ location }) {
  const { isAnimating, stopAnimation } = useContext(RouteTransitionAnimationContext);
  useLockBodyScroll(isAnimating);

  return isAnimating && <AnimatedOverlay location={location} onFinished={stopAnimation} />;
}

export default RouteTransitionAnimation;
