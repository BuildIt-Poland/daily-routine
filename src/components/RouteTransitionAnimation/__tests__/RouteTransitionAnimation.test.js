import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import { RouteTransitionAnimationContext } from '../../../context/RouteTransitionAnimationContext';
import RouteTransitionAnimation from '../RouteTransitionAnimation';

describe('COMPONENT - RouteTransitionAnimation', () => {
  it('renders correctly when "isAnimating" is  "true"', () => {
    const animateAndRedirect = () => {};
    const stopAnimation = () => {};
    const isAnimating = true;
    const theme = {
      primaryColor: '#fff',
      secondaryColor: '#222'
    };

    const component = create(
      <ThemeProvider theme={theme}>
        <RouteTransitionAnimationContext.Provider value={{ animateAndRedirect, isAnimating, stopAnimation }}>
          <RouteTransitionAnimation />
        </RouteTransitionAnimationContext.Provider>
      </ThemeProvider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
