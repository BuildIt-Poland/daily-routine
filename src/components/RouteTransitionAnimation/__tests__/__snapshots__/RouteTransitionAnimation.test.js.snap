// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - RouteTransitionAnimation renders correctly when "isAnimating" is  "true" 1`] = `
Array [
  .c0 {
  width: 100%;
  height: 160vh;
  background-color: #ffffff;
  left: 0;
  top: 6rem;
  position: absolute;
  z-index: 8000;
  opacity: 1;
  -webkit-transform: scaleY(0) translateY(0);
  -ms-transform: scaleY(0) translateY(0);
  transform: scaleY(0) translateY(0);
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

<div
    className="c0"
    style={
      Object {
        "transform": "scaleY(0)",
      }
    }
  />,
  .c0 {
  width: 100%;
  height: 0;
  left: 0;
  top: 6rem;
  position: absolute;
  z-index: 8002;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

<div
    className="c0"
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "height": "0%",
        "transform": "translateY(0vh)",
      }
    }
  />,
  .c0 {
  width: 100%;
  height: 160vh;
  background-color: #ffffff;
  left: 0;
  top: 6rem;
  position: absolute;
  z-index: 8000;
  opacity: 1;
  -webkit-transform: scaleY(0) translateY(0);
  -ms-transform: scaleY(0) translateY(0);
  transform: scaleY(0) translateY(0);
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  z-index: 8001;
  -webkit-transform-origin: 100% 100%;
  -ms-transform-origin: 100% 100%;
  transform-origin: 100% 100%;
}

<div
    className="c0"
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "opacity": 1,
        "transform": "scaleY(0) translateY(50%)",
      }
    }
  />,
]
`;
